api/src/test/java/com/epam/pipeline/controller/docker/DockerRegistryControllerTest.java

```diff
@@ -0,0 +1,311 @@
+    private final JwtRawToken jwtRawToken = SecurityCreatorUtils.getJwtRawToken();
+    private final DockerRegistryList dockerRegistryList = DockerCreatorUtils.getDockerRegistryList();
+    private final DockerRegistryEventEnvelope eventEnvelope = DockerCreatorUtils.getDockerRegistryEventEnvelope();
+
+    @Autowired
+    private DockerRegistryApiService mockDockerRegistryApiService;
+
+    @Autowired
+    private UserAccessService mockAccessService;
+
+    @Test
+    @WithMockUser
+    public void shouldCreateDockerRegistry() throws Exception {
+        final String content = getObjectMapper().writeValueAsString(dockerRegistryVO);
+        doReturn(dockerRegistry).when(mockDockerRegistryApiService).create(refEq(dockerRegistryVO));
+
+        final MvcResult mvcResult = performRequest(post(REGISTER_DOCKER_REGISTRY_URL).content(content));
+
+        verify(mockDockerRegistryApiService).create(refEq(dockerRegistryVO));
+        assertResponse(mvcResult, dockerRegistry, DockerCreatorUtils.DOCKER_REGISTRY_INSTANCE_TYPE);
+    }
+
+    @Test
+    public void shouldFailCreateDockerRegistry() throws Exception {
+        performUnauthorizedRequest(post(REGISTER_DOCKER_REGISTRY_URL));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldUpdateDockerRegistry() throws Exception {
+        final String content = getObjectMapper().writeValueAsString(dockerRegistry);
+        doReturn(dockerRegistry).when(mockDockerRegistryApiService).updateDockerRegistry(dockerRegistry);
+
+        final MvcResult mvcResult = performRequest(post(UPDATE_DOCKER_REGISTRY_URL).content(content));
+
+        verify(mockDockerRegistryApiService).updateDockerRegistry(dockerRegistry);
+        assertResponse(mvcResult, dockerRegistry, DockerCreatorUtils.DOCKER_REGISTRY_INSTANCE_TYPE);
+    }
+
+    @Test
+    public void shouldFailUpdateDockerRegistry() throws Exception {
+        performUnauthorizedRequest(post(UPDATE_DOCKER_REGISTRY_URL));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldUpdateDockerRegistryCredentials() throws Exception {
+        final String content = getObjectMapper().writeValueAsString(dockerRegistryVO);
+        doReturn(dockerRegistry).when(mockDockerRegistryApiService)
+                .updateDockerRegistryCredentials(refEq(dockerRegistryVO));
+
+        final MvcResult mvcResult = performRequest(post(UPDATE_DOCKER_REGISTRY_CREDS_URL).content(content));
+
+        verify(mockDockerRegistryApiService).updateDockerRegistryCredentials(refEq(dockerRegistryVO));
+        assertResponse(mvcResult, dockerRegistry, DockerCreatorUtils.DOCKER_REGISTRY_INSTANCE_TYPE);
+    }
+
+    @Test
+    public void shouldFailUpdateDockerRegistryCredentials() throws Exception {
+        performUnauthorizedRequest(post(UPDATE_DOCKER_REGISTRY_CREDS_URL));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldOauthEndpoint() throws Exception {
+        final UserContext userContext = SecurityCreatorUtils.getUserContext();
```

Comment:
It could be extracted even if there is only a single call now because it can be reused later.